src/lib/components/Swap/Settings/index.tsx

```diff
@@ -16,19 +16,26 @@ import TransactionTtlInput from './TransactionTtlInput'
 export function SettingsDialog() {
   const [boundary, setBoundary] = useState<HTMLDivElement | null>(null)
   const scrollbar = useScrollbar(boundary, { padded: true })
+
   const resetSettings = useResetAtom(settingsAtom)
+  const [maxSlippage, setMaxSlippage] = useState<ElementRef<typeof MaxSlippageSelect> | null>(null)
+  const onReset = useCallback(() => {
+    resetSettings()
+    maxSlippage?.reset()
+  }, [resetSettings, maxSlippage])
+
   return (
     <>
       <Header title={<Trans>Settings</Trans>} ruled>
-        <TextButton onClick={resetSettings}>
+        <TextButton onClick={onReset}>
           <ThemedText.ButtonSmall>
             <Trans>Reset</Trans>
           </ThemedText.ButtonSmall>
         </TextButton>
       </Header>
       <Column gap={1} style={{ paddingTop: '1em' }} ref={setBoundary} padded css={scrollbar}>
         <BoundaryProvider value={boundary}>
-          <MaxSlippageSelect />
+          <MaxSlippageSelect ref={setMaxSlippage} />
```

Comment:
Are you intending to set the ref here to the setState fn?

Reply:
Typings are broken when using `useRef` instead of `useState`, so I intentionally use `useState`.
If I were to use `useRef` I would need to coerce the type, and I'd rather not do that.
This pattern is also used in https://github.com/Uniswap/interface/blob/main/src/lib/components/TokenSelect/index.tsx#L63, for the same reasons.

Reply:
Is it not an option to fix the types?

Reply:
We can coerce the type, but it functions fine to just use state instead of a ref, and does so without subverting the type system, so I prefer it.
The types are broken in React, so it's more complex to fix the types than to work around them.